src/trading/getQuote.ts

```diff
@@ -104,20 +126,85 @@ export async function getQuote(
 
   log('Getting quote...')
 
-  const quoteResponse = await orderBookApi.getQuote(quoteRequest)
+  const quote = await orderBookApi.getQuote(quoteRequest)
+
+  // Get the suggested slippage based on the quote
+  const suggestedSlippageBps = suggestSlippageBps({ quote, tradeParameters, trader, advancedSettings })
+
+  // If no slippage is specified. AUTO slippage is used
+  if (slippageBps === undefined) {
+    // If suggested slippage is greater than default, we use the suggested slippage
+    if (suggestedSlippageBps > DEFAULT_SLIPPAGE_BPS) {
+      // Recursive call, this time using the suggested slippage
+      log(
+        `Suggested slippage is greater than ${DEFAULT_SLIPPAGE_BPS} BPS (default), using the suggested slippage (${suggestedSlippageBps} BPS)`
+      )
+
+      const newAppDataInfo = await buildAppData(
+        {
+          ...buildAppDataParams,
+          slippageBps: suggestedSlippageBps,
+        },
+        advancedSettings?.appData
+      )
+      log(
+        `App data with new suggested slippage: appDataKeccak256=${newAppDataInfo.appDataKeccak256} fullAppData=${newAppDataInfo.fullAppData}`
+      )
+
+      return {
+        slippageBps: suggestedSlippageBps,
+        suggestedSlippageBps,
+        tradeParameters: { ..._tradeParameters, slippageBps: suggestedSlippageBps },
+        appDataInfo: newAppDataInfo,
+
+        // We reuse the quote, because the slippage has no fundamental impact on the quote
+        quote,
+        orderBookApi,
+      }
+    } else {
+      log(
+        `Suggested slippage is only ${suggestedSlippageBps} BPS. Using the default slippage (${DEFAULT_SLIPPAGE_BPS} BPS)`
+      )
+    }
+  }
+
+  return {
+    quote,
+    appDataInfo,
+    orderBookApi,
+    tradeParameters,
+    slippageBps: slippageBpsOrDefault,
```

Comment:
Is this field meant to signify the slippage passed in as a parameter?
In that case, why return a default and not `undefined` when it isn't set?

Reply:
Good question. No what the user passed, they already know :) 
This signifies the ACTUAL slippage used. If `undefined`, it will return the suggested slippage instead  `line 155`. 

`slippageBpsOrDefault` will have either what the user inputed, or the default, in any case, its the slippage used. So we will always get a number informing the user the one we used.